### PR TITLE
added new interests in server

### LINKED ISSUE #22: 5. Add commands to Server.java
Please finish #21 first!
Add the 3 commands to Server.java and the code necessary to implement them.

### PR DESCRIPTION
**Pull request by mlk209**
Note: the commands that I used in server.java are found in lines 181 and 188 so, in network code the commands should be "NEW_INTERESTS_REQUEST" and "NEW_INTERESTS_RESPONSE". 

also, I think we might have to add to the controller.java file (line 186)? but im not completely sure? 

closes #22

### CODE REVIEW

**Comment by mlk209** [comment]
doesnt build, since im referring to files that @nvaa36  has written but not yet merged to the branch

**Comment by mlk209** [comment]
for line 189, should it be nullable or should it be collection? I'm assuming it should be collection from the server view java file?

File: src/codeu/chat/server/Server.java
Comment (nvaa36) [inline src/codeu/chat/server/Server.java]: Small issue, but this should be in alphabetic order for consistency.

File: src/codeu/chat/server/Server.java
Comment (nvaa36) [inline src/codeu/chat/server/Server.java]: Also minor, but I would put this with the other creation methods like newUser and newConversation.

File: src/codeu/chat/server/Server.java
Comment (nvaa36) [inline src/codeu/chat/server/Server.java]: The way that I have it implemented now and the way that I think it should be implemented (just slack me if you want to talk about it) is that for each of these server methods, instead of sending a collection of Uuids, you just send the 2 discrete Uuids, the user's Uuid and the interest's Uuid.
So to add an interest, you need 2 pieces of information, the user's Uuid and the interest's Uuid. So this block would look like:
```
final Uuid userid = Uuid.SERIALIZER.read(in);
final Uuid interest = Uuid.SERIALIZER.read(in);
```

File: src/codeu/chat/server/Server.java
Comment (nvaa36) [inline src/codeu/chat/server/Server.java]: There's also no way for you to know, but I named this `newInterest(Uuid userid, Uuid interest)` not newInterests(Set<Uuids> ids) because you're adding a single interest, not an Interests object necessarily.

**Review by nvaa36 — CHANGES_REQUESTED** [review]
Ok, looks mostly good, just a few minor changes. Also, are you going to add the other methods?

File: src/codeu/chat/server/Server.java
Comment (mlk209) [reply src/codeu/chat/server/Server.java]: got it

File: src/codeu/chat/server/Server.java
Comment (mlk209) [reply src/codeu/chat/server/Server.java]: okay, fixed it!

File: src/codeu/chat/server/Server.java
Comment (mlk209) [reply src/codeu/chat/server/Server.java]: added it in!

File: src/codeu/chat/server/Server.java
Comment (mlk209) [reply src/codeu/chat/server/Server.java]: changed it!

File: src/codeu/chat/server/Server.java
Comment (nvaa36) [inline src/codeu/chat/server/Server.java]: May have steered you wrong a little bit here, but I think that this should be `newInterest(userid, interest)`, you generally don't have the type when you're calling a function, only when you are defining one.

**Review by nvaa36 — CHANGES_REQUESTED** [review]
Also, are you adding a getInterests and a getInterestsByUserId?

File: src/codeu/chat/server/Server.java
Comment (mlk209) [reply src/codeu/chat/server/Server.java]: Okay i just fixed it, the changes will be in the next commit!

**Comment by mlk209** [comment]
I think i added the all the methods now? lmk what needs to be changed?

File: src/codeu/chat/server/Server.java
Comment (nvaa36) [inline src/codeu/chat/server/Server.java]: This should be `interests` because you're returning the full Interests object for the user.

File: src/codeu/chat/server/Server.java
Comment (nvaa36) [inline src/codeu/chat/server/Server.java]: `Interests` shouldn't be capitalized when passed in as a parameter.

File: src/codeu/chat/server/Server.java
Comment (nvaa36) [inline src/codeu/chat/server/Server.java]: This function, as written, only allows the client to get a single Interests object for a single user, I think it should be able to get a Collection of them to match the functionality of the other get functions, but if not, then just change the comment so it reflects the actual behavior.

**Review by nvaa36 — CHANGES_REQUESTED** [review]
Ok, so a few minor changes, and then just wait until the previous issues are merged, check to make sure this one builds correctly, then we can merge it.

**Review by nvaa36 — APPROVED** [review]

**Comment by nvaa36** [comment]
Ok, looks good as long as the build passes!

**Pull request merged**
